### PR TITLE
Add placeholders for targeted content translations

### PR DESCRIPTION
**Pull request by davidrapson**
Add placeholders for targeted content translations. By default Rails injects translation missing markup, which when used in attributes results in the following:

Screenshot 2020-12-16 at 13 34 27

Adding placeholders to cy.yml for now, have requested translations but imagine they will come in the new year so suggest merging this as is.

### CODE REVIEW

**Review by baseonmars — APPROVED** [review]
Looking good!

Given our use case do you think it would be worth setting up fallbacks in the rails app?

alternatively - we should have a script that compares the two files and reports any missing keys as part of CI.

**Comment by davidrapson** [comment]
> alternatively - we should have a script that compares the two files and reports any missing keys as part of CI.

I'd be up for this. Normally you can set `raise_on_missing_translations` in development to catch this, which would be possible if we went [full engine](https://github.com/citizensadvice/design-system/pull/594), but couldn't get that working in our script context so a CI check would be a good safeguard.

**Comment by davidrapson** [comment]
I've updated this PR with a basic lint script which compares the keys in both locale files and should error if anything is missing

File: Rakefile [hunk 1]
```diff
@@ -18,6 +19,19 @@ def base_cucumber_path
   "artifacts/#{ENV['BROWSER']}/#{ENV['BROWSERSTACK'] == 'true' ? ENV['BROWSERSTACK_CONFIGURATION_OPTIONS'] : 'other'}"
 end
 
+def compare_yaml_hash(yaml_a, yaml_b, context = [])
+  yaml_a.each do |key, value|
```
Comment (luke-hill) [inline Rakefile:23]: If i'm understanding here, you're wanting to validate the nested structure of the yaml, but only validate the keys right?

Comment (davidrapson) [reply Rakefile:23]: Yep, want to validate that all the keys match. But not the values obviously as they are translated. Not looking for a perfect check, only a little safeguard. Simpler ways to do this welcome.

**Comment by davidrapson** [comment]
Hmm this _does_ fail as expected but `collect_task_errors` seems to swallow the actual error message in CI

File: Rakefile (hunk 1 above)
Comment (luke-hill) [reply Rakefile:23]: Actually it seems quite good. Although I saw it's a SO answer that you've based it off. Only thing I got confused by was the name. It's definitely a **very good** way of using clever recursion. I would have suggested `hashdiff` but that computes value difference too using diff-lcs as an algorithm (Which you obviously won't want), so this is a good solution.

If it was me I'd call it something like `ensure_identical_structure(yml_a, yml_b, context = [])` but that's definitely just a personal opinion. Not something I'm fussed on too much

File: Gemfile.lock [hunk 2]
```diff
@@ -215,6 +217,7 @@ DEPENDENCIES
   rake
   retriable (~> 3.1)
   rspec
+  rubocop-rake
```
Comment (luke-hill) [inline Gemfile.lock:220]: This is unlikely to run by default, it'll probably need configuring in the ca style repo proper. Or patched in to our config in this repo inside the `.rubocop.yml` configuration file

NB: Disclaimer (I've never used this extension, so I could be wrong!)

File: Rakefile (hunk 1 above)
Comment (davidrapson) [reply Rakefile:23]: [looks around shiftily] definitely did~n't~ copy paste this guvner.

File: Gemfile.lock (hunk 2 above)
Comment (davidrapson) [reply Gemfile.lock:220]: There's a `require: rubocop-rake` in the .rubocop.yml if you missed it. Caught a couple little things and as I was adding more to the rakefile it felt helpful.

**Comment by davidrapson** [comment]
Closing in favour of https://github.com/citizensadvice/design-system/pull/681

**Pull request closed without merging**